snuba/migrations/snuba_migrations/metrics/0025_metrics_counters_aggregate_v2.py

```diff
@@ -0,0 +1,99 @@
+from typing import Sequence
+
+from snuba.clickhouse.columns import Array, Column, UInt
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.datasets.storages.tags_hash_map import INT_TAGS_HASH_MAP_COLUMN
+from snuba.migrations import migration, operations, table_engines
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+from snuba.migrations.snuba_migrations.metrics.templates import COMMON_AGGR_COLUMNS
+from snuba.utils.schemas import AggregateFunction, Float, String
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    blocking = False
+    table_name = "metrics_counters_v2_local"
+    dist_table_name = "metrics_counters_v2_dist"
+    granularity = "1024"
+    aggregated_cols: Sequence[Column[Modifiers]] = [
+        # This is a common column for aggregate tables going forward
+        # but doesn't exist in the old templates so we include it here
+        Column("use_case_id", String(Modifiers(low_cardinality=True))),
+        *COMMON_AGGR_COLUMNS,
+        Column("value", AggregateFunction("sum", [Float(64)])),
+    ]
+
+    def forwards_local(self) -> Sequence[operations.SqlOperation]:
+
+        return [
+            operations.CreateTable(
+                storage_set=StorageSetKey.METRICS,
+                table_name=self.table_name,
+                columns=self.aggregated_cols,
+                engine=table_engines.AggregatingMergeTree(
+                    storage_set=StorageSetKey.METRICS,
+                    order_by="(org_id, project_id, metric_id, granularity, timestamp, tags.key, tags.value)",
```

Comment:
We need to separate the ORDER BY key from the Primary key

Like this 
```
PRIMARY KEY (org_id, project_id, metric_id, granularity, timestamp)
ORDER BY (org_id, project_id, metric_id, granularity, timestamp, tags.key, tags.value)
```

This ensures we store value uniquely by ORDER BY and sorted that way but we include only `org_id, project_id, metric_id, granularity, timestamp` in the index, which reduces the index size.

Also, please add the use_case_id to the ORDER BY key as well (either before or after tags) but not to the PRIMARY KEY. The partition key should be enough to prevent Clickhosue from merging rows of different retentions but we never know.